File: Problem 007/Solution 7.rb
# Solves http://projecteuler.net/problem=7
require '../includes/Ruby/Prime.rb'

primecount = 1
number = 3

while(primecount != 10001)
	primecount += 1 if(IsPrime(number) == 1)
	number += 2 if (primecount != 10001)
end

print "Answer: ", number, "\n"

File: Problem 007/Solution 7.c
/* Solves http://projecteuler.net/problem=7 */
#include <stdio.h>
#include "../includes/C/Prime.h"

int main()
{
	/* Set vars */
	int primecount, number;

	/* Loop until we break */
	for(primecount = 0, number = 1; primecount != 10001; number += 2)
	{
		/* If we find a prime, incrase x
			if x is 10001, break the loop */
		if(IsPrime(number) == 1)
			++primecount;

		/* If primecount is 10001 then break the loop. */
		if(primecount == 10001)
			break;

	}

	/* Print the answer */
	printf("Answer: %d\r\n", number);

	return 0;
}
